src/MaxText/input_pipeline/_input_pipeline_utils.py

```diff
@@ -389,6 +389,16 @@ def map(self, element):
       return {col: element[col].numpy() for col in self.column_names}
 
 
+@dataclasses.dataclass
+class KeepFeatures(grain.MapTransform):
+
+  def __init__(self, feature_names):
+    self.feature_names = feature_names
+
+  def map(self, element):
+    return {k: v for k, v in element.items() if k in self.feature_names}
```

Comment:
Could we add typing to the arguments / return types of this function?

Reply:
done